client/docs/concepts/lifecycle.md

```diff
@@ -0,0 +1,48 @@
+# Job Lifecycle
+
+Since jobq builds on top of the [Kueue](https://kueue.sigs.k8s.io/) job queuing system for scheduling,
+the lifecycle of a job is very similar to the lifecycle of a workload in Kueue.
+
+The remainder of this document uses the terms _job_ and _workload_ interchangeably.
+
+A workload roughly goes through three phases after its submission: _queuing and scheduling_, _execution_, and _completion_.
+
+### Queueing and scheduling
+
+After its submission, a workload is in the `Submitted` state, where it competes with other workloads for available resource quotas.
+Once it is admitted to a cluster queue, it enters the `Pending` state, where Kueue will reserve a quota for it.
+Alternatively, if the selected local or cluster queue for the workload are stopped, the workload will enter the `Inadmissible` state until this condition is resolved.
+
+### Execution
+
+After all admission checks for the workload have passed, it enters the `Admitted` state, it is now eligible for execution by the cluster.
+
+### Completion
+
+When the workload terminates successfully, it enters the terminal `Succeeded` state.
+If any unrecoverable error occurs during execution, the workload enters the terminal `Failed` state.
```

Comment:
Maybe it's worth highlighting that depending on the completion/backoff policy, "my pod has failed" does not equal "my job has failed". (Thinking back to our list pair programming session)

Reply:
Improved wording in 4d1579d892f8d9e8027539d3021d056f7c461bf0